src/coreComponents/mesh/MeshManager.cpp

```diff
@@ -75,13 +79,86 @@ void MeshManager::generateMeshLevels( DomainPartition & domain )
   } );
 }
 
-void MeshManager::importFields( DomainPartition & domain )
+/**
+ * @brief Collect a set of material field names registered in a subregion.
+ * @param subRegion the target subregion
+ * @return a set of wrapper names
+ */
+std::unordered_set< string > getMaterialWrapperNames( ElementSubRegionBase const & subRegion )
 {
-  forSubGroups< MeshGeneratorBase >( [&]( MeshGeneratorBase & meshGen )
+  using namespace constitutive;
+  std::unordered_set< string > materialWrapperNames;
+  subRegion.getConstitutiveModels().forSubGroups< ConstitutiveBase >( [&]( ConstitutiveBase const & material )
   {
-    meshGen.importFields( domain );
+    material.forWrappers( [&]( WrapperBase const & wrapper )
+    {
+      if( wrapper.sizedFromParent() )
+      {
+        materialWrapperNames.insert( ConstitutiveBase::makeFieldName( material.getName(), wrapper.getName() ) );
+      }
+    } );
   } );
+  return materialWrapperNames;
 }
 
+void MeshManager::importFields( DomainPartition & domain )
+{
+  GEOSX_MARK_FUNCTION;
+  forSubGroups< MeshGeneratorBase >( [&]( MeshGeneratorBase & generator )
+  {
+    try
+    {
+      VTKMeshGenerator & g = dynamic_cast< VTKMeshGenerator & >( generator );
+      GEOSX_UNUSED_VAR( g );
+    }
+    catch( const std::bad_cast & e )
+    {
+      return;
+    }
```

Comment:
Why do you only work on `VTKMeshGenerator` and not on all `MeshGeneratorBase`s ?
I have a few comments about it.

- If we really wanted to do this, we would use some `forSubGroups< VTKMeshGenerator >` instead.
- Please use our `dynamicCast` (which may lead to a tiny differente behavior)
- But more importantly, by doing so, you expose the implementation `VTKMeshGenerator` which eventually should not be visible from outside of the `mesh` module.

I think that you could loop on all `MeshGeneratorBase` instances. Instances with no particular data to import will return empty dummy values (like `fieldNamesMapping` will be an empty `map`), and all the import loops will vanish. We'll remove a `if` in the code which is always nice. And the `resqml` guys won't have to modify anything from this code if they implement their own `MeshGeneratorBase`.

Reply:
Maybe implement an equivalent to `domain.hasMeshBody(...)`?